debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/dialect/GeneralDatabaseDialect.java

```diff
@@ -783,6 +788,17 @@ private String columnNameEqualsBinding(String fieldName, TableDescriptor table,
         return toIdentifier(columnName) + "=" + field.getQueryBinding(column, record.getAfterStruct());
     }
 
+    private String whereColumnNameEqualsBinding(String fieldName, TableDescriptor table, SinkRecordDescriptor record) {
+        final FieldDescriptor field = record.getFields().get(fieldName);
+        final String columnName = resolveColumnName(field);
+        final ColumnDescriptor column = table.getColumnByName(columnName);
+        final Object value = Optional.ofNullable(record.getKeyStruct(connectorConfig.getPrimaryKeyMode()))
+                .map(it -> it.get(fieldName))
+                .orElse(null);
```

Comment:
```suggestion
        final Object value;
        if (record.getNonKeyFieldNames().contains(fieldName)) {
            value = getColumnValueFromValueField(fieldName, record);
        }
        else {
            value = getColumnValueFromKeyField(fieldName, record, columnName);
        }
```

в классе есть готовые методы для получения значения из ключевых и неключевых столбцов (смотрел на метод columnQueryBindingFromField

и чтобы не фиксировать жестко в методе откуда идет поле (из набор ключевых столбцов или из набора неключевых - это может меняться настройка на уровнях выше), предлагаю оттуда взять логику на получения значения

можно даже в отдельный метод вынести чтобы дублирование убрать